packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -666,7 +683,15 @@ export class Bucket extends BucketBase {
       bucketEncryption,
       versioningConfiguration: props.versioned ? { status: 'Enabled' } : undefined,
       lifecycleConfiguration: new cdk.Token(() => this.parseLifecycleConfiguration()),
-      websiteConfiguration: this.renderWebsiteConfiguration(props)
+      websiteConfiguration: this.renderWebsiteConfiguration(props),
+      publicAccessBlockConfiguration: props.blockPublicAccess
```

Comment:
We should support the entire surface area of the feature, but also find a way to provide a more ergonomic API.

How about using an "enum-like class":

```ts
export interface BlockPublicAccessOptions {
  blockPublicAcls?: boolean;
  ignorePublicAcls?: boolean;
  blockPublicPolicy?: boolean;
  restrictPublicBuckets?: boolean;
}

export class BlockPublicAccess {
  public static blockAll = new BlockPublicAccess({ blockPublicAcls: true, ...true, true, true... });
  public static block... // maybe there are other canned settings we can provide

  constructor(options: BlockPublicAccessOptions) { ... }
}
```

Reply:
@jogold have you seen this comment?

Reply:
We try not to "leak" the CFN layer, eventually L2 should cover the entire surface area. We don't want people to need to traverse these layers to use prop overrides. Please define another props interface. In most cases it's not a 1:1 mapping (although it might be here). We also want to able to evolve those independently.

As for naming, `blockPublicAccess` of type `BlockPublicAccess`, so the usage would be:

```ts
blockPublicAccess: BlockPublicAccess.blockAll
```

(a bit of a mouthfull but not too bad)

Reply:
See latest commit:
- used `BlockAll` (capitalized)
- added `BlockAcls`
- `grantPublicRead` now only throws when `blockPublicPolicy` is enabled because it's only in this case that S3 rejects calls to PUT Bucket policy.